src/System.Collections/src/System/Collections/Generic/HashSetEqualityComparer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Collections.Generic
+{
+
+    /// <summary>
+    /// Equality comparer for hashsets of hashsets
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal class HashSetEqualityComparer<T> : IEqualityComparer<HashSet<T>>
+    {
+
+        private IEqualityComparer<T> m_comparer;
```

Comment:
Nit: This should be named `_comparer` and can be `readonly`. Also, the extra blank line above can be removed.